index.go

```diff
@@ -1,47 +1,2085 @@
+			img = mutate.Annotations(img, annos).(v1.Image)
+		}
+
+		if len(mfest.Config.URLs) != 0 {
+			ops = append(ops, layout.WithURLs(mfest.Config.URLs))
+		}
+
+		if platform := mfest.Config.Platform; platform == nil || platform.Equals(v1.Platform{}) {
+			if platform == nil {
+				platform = &v1.Platform{}
+			}
+
+			config, err := img.ConfigFile()
+			if err != nil {
+				return err
+			}
+
+			if config == nil {
+				return ErrConfigFileUndefined
+			}
+
+			if err = updatePlatform(config, platform); err != nil {
+				return err
+			}
+
+			ops = append(ops, layout.WithPlatform(*platform))
+		}
+
+		i.Images[desc.Digest] = img
+		iMap.Store(img, ops)
+		return nil
 	default:
-		return ""
+		return ErrUnknownMediaType
+	}
+}
+
+func addPlatformSpecificImages(i *Index, ref name.Reference, platform v1.Platform, annotations map[string]string) error {
+	if platform.OS == "" || platform.Architecture == "" {
+		return ErrInvalidPlatform
+	}
+
+	desc, err := remote.Get(
+		ref,
+		remote.WithAuthFromKeychain(i.Options.KeyChain),
+		remote.WithTransport(getTransport(true)),
+		remote.WithPlatform(platform),
+	)
+	if err != nil {
+		return err
+	}
+
+	return appendImage(i, desc, annotations)
+}
+
+func appendImage(i *Index, desc *remote.Descriptor, annotations map[string]string) error {
+	img, err := desc.Image()
+	if err != nil {
+		return err
+	}
+
+	digest, err := img.Digest()
+	if err != nil {
+		return err
+	}
+
+	mfest, err := img.Manifest()
+	if err != nil {
+		return err
+	}
+
+	if mfest == nil {
+		return ErrManifestUndefined
+	}
+
+	var layoutOps []layout.Option
+	var annos = make(map[string]string)
+	if len(annotations) != 0 && mfest.MediaType == types.OCIManifestSchema1 {
+		if len(mfest.Annotations) != 0 {
+			annos = mfest.Annotations
+		}
+
+		for k, v := range annotations {
+			annos[k] = v
+		}
+
+		layoutOps = append(layoutOps, layout.WithAnnotations(annos))
+		// i.Annotate.SetAnnotations(digest, annos)
+		// i.Annotate.SetFormat(digest, desc.MediaType)
+		img = mutate.Annotations(img, annos).(v1.Image)
+	}
+
+	if len(mfest.Config.URLs) != 0 {
+		layoutOps = append(layoutOps, layout.WithURLs(mfest.Config.URLs))
+	}
+
+	if platform := mfest.Config.Platform; platform == nil || platform.Equals(v1.Platform{}) {
+		if platform == nil {
+			platform = &v1.Platform{}
+		}
+
+		config, err := img.ConfigFile()
+		if err != nil {
+			return err
+		}
+
+		if config == nil {
+			return ErrConfigFileUndefined
+		}
+
+		if err = updatePlatform(config, platform); err != nil {
+			return err
+		}
+
+		layoutOps = append(layoutOps, layout.WithPlatform(*platform))
+	}
+
+	layoutPath := filepath.Join(i.Options.XdgPath, i.Options.Reponame)
+	path, err := layout.FromPath(layoutPath)
+	if err != nil {
+		path, err = layout.Write(layoutPath, i.ImageIndex)
+		if err != nil {
+			return err
+		}
+	}
+
+	i.Images[digest] = img
+	return path.AppendImage(img, layoutOps...)
+}
+
+func (i *Index) Save() error {
```

Comment:
I am confused about how this method is working. After I tested the `manifest create` method I can see in my PACK_HOME that the whole Image was saved there in OCI layout format. 


<img width="1792" alt="Screenshot 2024-02-09 at 10 30 14 AM" src="https://github.com/husni-faiz/imgutil/assets/1181799/40bf185a-d70c-4278-a3d8-929f17a05472">

Why do we need the whole image? Image I am creating an Index for images that are X GB in size, are we going to write on disk all the blobs for those images ?